build/AnalyzeDependencies/AnalyzeDepsJS.ps1

```diff
@@ -0,0 +1,261 @@
+#
+
+<#
+  .SYNOPSIS
+  Generate a dependency report for a set of .tgz files
+  .PARAMETER PackagesPath
+  The path to the package(s) to analyze. Globs are supported.
+  .PARAMETER LockfilePath
+  NOT USED. The path to the lockfile to analyze.
+  .PARAMETER OutPath
+  The path to write the HTML-formatted report.
+  .PARAMETER DumpPath
+  The path to write the JSONP-formatted dependency data file.
+#>
+
+Param(
+  [Parameter(Mandatory = $true)][string]$PackagesPath,
+  [Parameter(Mandatory = $false)][string]$LockfilePath,
+  [Parameter(Mandatory = $false)][string]$OutPath,
+  [Parameter(Mandatory = $false)][string]$DumpPath
+)
+
+Add-Type -AssemblyName System.IO.Compression.FileSystem
+
+Function Get-PackageJson($NupkgPath) {
+  try {
+    # Expand tar file to tempdir folder
+    Remove-Item -Path ./tempdir -Recurse -Force -ErrorAction SilentlyContinue
+    New-Item -Path ./tempdir -ItemType directory -Force | Out-Null;
+
+    Expand-Tar $NupkgPath ./tempdir
+    #pushd ./tempdir
+    #tar -xvzf $NupkgPath >$null 2>&1;
+    $ZipFile = (Get-ChildItem package.json -Path ./tempdir -Recurse ).FullName;
+    #popd
+
+    Get-Content -Raw $ZipFile | ConvertFrom-Json;
+  }
+  catch {
+  }
+}
+
+Function Expand-Tar($tarFile, $dest) {
+    if (-not (Get-Command Expand-7Zip -ErrorAction Ignore)) {
+        Install-Package -Scope CurrentUser -Force 7Zip4PowerShell > $null
+    }
+    Expand-7Zip $tarFile $dest
+    # Expand the inner tar file
+    $tar = Get-ChildItem $dest/*.tar
+    if ($tar) { Expand-7Zip $tar $dest; }
+}
+
+Function Save-Dep($Deps, $TargetFramework, $DepName, $DepVersion, $DependentPackage) {
+  if (-Not $Deps[$DepName]) {
+    $Deps[$DepName] = @{ }
+  }
+  if (-Not $Deps[$DepName][$TargetFramework]) {
+    $Deps[$DepName][$TargetFramework] = @{ }
+  }
+  if (-Not $Deps[$DepName][$TargetFramework][$DepVersion]) {
+    $Deps[$DepName][$TargetFramework][$DepVersion] = New-Object System.Collections.ArrayList
+  }
+  if (-Not $Deps[$DepName][$TargetFramework][$DepVersion].Contains($DependentPackage)) {
+    $Deps[$DepName][$TargetFramework][$DepVersion].Add($DependentPackage) | Out-Null
+  }
+}
+
+Function Save-PkgDep($PkgDeps, $TargetFramework, $DepName, $DepVersion) {
+  if (-Not $PkgDeps[$DepName]) {
+    $PkgDeps[$DepName] = @{ }
+  }
+  if (-Not $PkgDeps[$DepName][$DepVersion]) {
+    $PkgDeps[$DepName][$DepVersion] = New-Object System.Collections.ArrayList
+  }
+  if (-Not $PkgDeps[$DepName][$DepVersion].Contains($TargetFramework)) {
+    $PkgDeps[$DepName][$DepVersion].Add($TargetFramework) | Out-Null
+  }
+}
+
+#Function Save-Locked($Locked, $DepName, $DepVersion, $Condition) {
+#  if (-Not $Locked[$DepName]) {
+#    $Locked[$DepName] = @{ }
+#  }
+#  if (-Not $Locked[$DepName][$DepVersion]) {
+#    $Locked[$DepName][$DepVersion] = New-Object System.Collections.ArrayList
+#  }
+#  if (-Not $Locked[$DepName][$DepVersion].Contains($Condition)) {
+#    $Locked[$DepName][$DepVersion].Add($Condition.Trim()) | Out-Null
+#  }
+#}
```

Comment:
Is this necessary? When would we uncomment the function `Save-Locked`?

Reply:
I guess not. Deleting...